protocol.cpp

```diff
@@ -36,7 +36,7 @@ CMessageHeader::CMessageHeader()
 CMessageHeader::CMessageHeader(const char* pszCommand, unsigned int nMessageSizeIn)
 {
     memcpy(pchMessageStart, ::pchMessageStart, sizeof(pchMessageStart));
-    int command_len = strlen(pszCommand);
+    int command_len = strnlen(pszCommand, COMMAND_SIZE);
```

Comment:
I suggest also changing the type to `size_t` to match `strnlen`

Reply:
Done.